lib/rule-doc-notices.ts

```diff
@@ -197,6 +199,43 @@ const RULE_NOTICES: {
     ruleName,
     urlRuleDoc,
   }) => {
+    // use object type `DeprecatedInfo`
+    if (typeof deprecatedInfo === 'object') {
+      const replacementRuleList = (deprecatedInfo.replacedBy ?? [])
+        .map(({ rule }) =>
+          rule && rule.name
+            ? rule.url
+              ? `[\`${rule.name}\`](${rule.url})`
+              : `\`${rule.name}\``
+            : undefined,
+        )
+        .filter((rule): rule is string => typeof rule === 'string');
+
+      return `${EMOJI_DEPRECATED} This rule is deprecated${
+        deprecatedInfo.deprecatedSince
+          ? ` since v${deprecatedInfo.deprecatedSince}.`
+          : '.'
+      }${
+        replacementRuleList.length > 0
+          ? ` It was replaced by ${String(replacementRuleList)}.`
+          : ''
+      }${
+        // use DeprecatedInfo#url to inform about the reasons
+        deprecatedInfo.url
+          ? `${EOL}${EOL}Read more at [${new URL(deprecatedInfo.url).hostname}](${deprecatedInfo.url})`
```

Comment:
It's an interesting idea to use the hostname, but I think I'd rather just use this, and keep everything related to deprecations on the same line.
```suggestion
          ? `[Read more](deprecatedInfo.url).`
```